tools/jolie2dummy/src/joliex/dummycreator/impl/JolieDummyDocumentCreator.java

```diff
@@ -120,239 +122,51 @@ private void convertOperation( OperationDeclaration operatationDeclaration, Stri
 	{
 
 		if ( operatationDeclaration instanceof RequestResponseOperationDeclaration ) {
-			//if (((RequestResponseOperationDeclaration)operatationDeclaration).requestType().id()
 			stringBuilder.append( "[" ).append( operatationDeclaration.id() ).append( "(request)(response){\n" );
 			stringBuilder.append( "\t" ).append( "valueToPrettyString@StringUtils( request )( s)" ).append( ";\n" );
-			stringBuilder.append( "\t" ).append( "println@Console(s)()" );
+			stringBuilder.append( "\t" ).append( "println@Console(s)();\n" );
 
 			TypeDefinition typeDefinition = ((RequestResponseOperationDeclaration) operatationDeclaration).responseType();
-			if ( typeDefinition.hasSubTypes() ) {
-				stringBuilder.append( ";\n" );
-				for( Entry<String, TypeDefinition> entry : typeDefinition.subTypes() ) {
-
-					convertTypes( entry.getValue(), stringBuilder, "\tdummyResponse." + entry.getKey() );
-
-
-				}
-				stringBuilder.append( "response<<dummyResponse\n" );
-			} else {
-
-				if ( !(typeDefinition.nativeType().id().equals( "void" )) ) {
-					stringBuilder.append( ";\n" );
-					if ( typeDefinition.nativeType().id().equals( "int" ) ) {
-						stringBuilder.append( "\tdummyRespose" ).append( "=" ).append( "42;\n" );
-//
-					} else if ( typeDefinition.nativeType().id().equals( "double" ) ) {
-
-						stringBuilder.append( "\tdummyRespose" ).append( "=" ).append( "1.54;\n" );
-
-
-					} else {
-
-						stringBuilder.append( "\tdummyRespose" ).append( "=" ).append( "\"dummy" ).append( typeDefinition.id() ).append( "\"" ).append( ";\n" );
-
-
-					}
-					stringBuilder.append( "response<<dummyResponse\n" );
-				}
-
-			}
-
-			/// handeling of types here
-
+			convertTypes( typeDefinition, stringBuilder, "dummyResponse");
+			stringBuilder.append("response<<dummyResponse\n" );
 			stringBuilder.append( "\n}]{nullProcess}\n" );
-
 		}
 
 	}
 
-	private void convertTypes( TypeDefinition typeDefinition, StringBuilder stringBuilder, String nameVariable )
-	{
-
-
-		if ( typeDefinition instanceof TypeDefinitionLink ) {
-
+	private void convertTypes( TypeDefinition typeDefinition, StringBuilder stringBuilder, String nameVariable) {
+		if (typeDefinition instanceof TypeChoiceDefinition) {
+			TypeChoiceDefinition typeChoiceDefinition = (TypeChoiceDefinition) typeDefinition;
+			convertTypes(typeChoiceDefinition.left(), stringBuilder, nameVariable);
+			convertTypes(typeChoiceDefinition.right(), stringBuilder, nameVariable);
```

Comment:
Mmh I think that this line should be erased. Assume that we have this type:

type t : int | string

Then your implementation would generate something like:

x = 42;
x = "dummy"

right? The second one is redundant, as it overwrites the first.